examples/models/file/gauges.py

```diff
@@ -76,25 +75,25 @@ def add_gauge(radius, max_value, length, direction, color, major_step, minor_ste
 
     rotation = 0 if direction == 1 else -pi
 
-    x, y = zip(*[ polar_to_cartesian(radius, angle) for angle in major_angles ])
     angles = [ angle + rotation for angle in major_angles ]
-    source = ColumnDataSource(dict(x=x, y=y, angle=angles))
+    source = ColumnDataSource(dict(major_angles=major_angles, angle=angles))
 
-    glyph = Ray(x="x", y="y", length=data(length), angle="angle", line_color=color, line_width=2)
+    t = PolarTransform(radius=radius, angle="major_angles")
+    glyph = Ray(x=expr(t.x), y=expr(t.y), length=data(length), angle="angle", line_color=color, line_width=2)
     plot.add_glyph(source, glyph)
 
-    x, y = zip(*[ polar_to_cartesian(radius, angle) for angle in minor_angles ])
     angles = [ angle + rotation for angle in minor_angles ]
-    source = ColumnDataSource(dict(x=x, y=y, angle=angles))
+    source = ColumnDataSource(dict(minor_angles=minor_angles, angle=angles))
 
-    glyph = Ray(x="x", y="y", length=data(length/2), angle="angle", line_color=color, line_width=1)
+    t = PolarTransform(radius=radius, angle="minor_angles")
+    glyph = Ray(x=expr(t.x), y=expr(t.y), length=data(length/2), angle="angle", line_color=color, line_width=1)
     plot.add_glyph(source, glyph)
 
-    x, y = zip(*[ polar_to_cartesian(radius+2*length*direction, angle) for angle in major_angles ])
     text_angles = [ angle - pi/2 for angle in major_angles ]
-    source = ColumnDataSource(dict(x=x, y=y, angle=text_angles, text=major_labels))
+    source = ColumnDataSource(dict(major_angles=major_angles, angle=text_angles, text=major_labels))
 
-    glyph = Text(x="x", y="y", angle="angle", text="text", text_align="center", text_baseline="middle")
+    t = PolarTransform(radius=radius + 2*length*direction, angle="major_angles")
+    glyph = Text(x=expr(t.x), y=expr(t.y), angle="angle", text="text", text_align="center", text_baseline="middle")
```

Comment:
are the `expr` here necessary?

Reply:
It's not, but as this is a model-level API example, then it's fine here.